src/components/internal/Card/Card.tsx

```diff
@@ -31,6 +39,35 @@ const Card = ({
     testId,
 }: PropsWithChildren<CardProps>) => {
     const Tag = el || 'header';
+    const [showContent, setShowContent] = useState(false);
+    const cardId = useMemo(() => uuid(), []);
+
+    const toggleExpansion = useCallback(() => {
+        if (expandable) {
+            setShowContent(!showContent);
+        }
+    }, [expandable, showContent]);
+
+    const onKeyDown = useCallback(
+        (evt: KeyboardEvent) => {
+            if (evt.code === InteractionKeyCode.ENTER || evt.code === InteractionKeyCode.SPACE) {
+                toggleExpansion();
+            }
```

Comment:
I'd suggest you also prevent default action here.

```suggestion
            switch (evt.code) {
                case InteractionKeyCode.ENTER:
                case InteractionKeyCode.SPACE:
                    evt.preventDefault();
                    toggleExpansion();
                    return;
            }
```